src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -175,9 +191,24 @@ private async Task ProcessQueueAsync()
                 try
                 {
                     var (work, activityId, cancellationToken) = queueItem;
+                    CancellationTokenSource? cancellationTokenSource = null;
                     lspServices = work.LspServices;
 
-                    var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(CancellationToken, cancellationToken);
+                    if (CancelInProgressWorkUponMutatingRequest)
+                    {
+                        // Verify this queueitem hasn't already been cancelled before creating a linked
+                        // CancellationTokenSource based on it.
+                        if (cancellationToken.IsCancellationRequested)
+                        {
+                            cancellationToken.ThrowIfCancellationRequested();
+                        }
+
+                        cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(CancellationToken, cancellationToken);
```

Comment:
do we need this?  The request `cancellationToken` is already linked to the queue `CancellationToken` unless I'm mis-reading.  As an aside I would suggest potentially renaming some of these to be specific to what the cancellation token represents

Reply:
I've renamed some of these things to hopefully make it more clear. Can you take another look and see if it still doesn't make sense? I think I need a cancellationtokensource for the current work if I need to cancel it later.

Reply:
Got rid of the throw and added an ObjectDisposedException handler to handle this case.